src/core/payments/types.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Service approval status from the Payments contract
+ */
+export interface ServiceApprovalStatus {
+  rateAllowance: bigint
+  lockupAllowance: bigint
+  lockupUsed: bigint
+  maxLockupPeriod?: bigint
+  rateUsed?: bigint
```

Comment:
```suggestion
  maxLockupPeriod: bigint
  rateUsed: bigint
```

these are not optional in synapse so it looks like we can make them mandatory here, there's one test in filecoin-pin that doesn't set them, but other than that I think it compiles with them set to mandatory